src/tasks/WasmAppBuilder/WasmAppBuilder.cs

```diff
@@ -408,4 +449,43 @@ private static bool TryConvert(string str, Type type, out object? value)
             return false;
         }
     }
+
+    public static bool IsAssembly(byte[] bytes)
+    {
+        try
+        {
+            // Try to read CLI metadata from the PE file.
+            using var peReader = new PEReader(ImmutableArray.Create(bytes));
+
+            if (!peReader.HasMetadata)
+            {
+                return false; // File does not have CLI metadata.
+            }
+
+            // Check that file has an assembly manifest.
+            MetadataReader reader = peReader.GetMetadataReader();
+            return reader.IsAssembly;
+        }
+        catch (BadImageFormatException)
+        {
+            return false;
+        }
+        catch (FileNotFoundException)
+        {
+            return false;
+        }
+    }
+
+    public static string GetFileHash(string filePath)
+    {
+        return GetFileHash(File.ReadAllBytes(filePath));
+    }
+
+    public static string GetFileHash(byte[] bytes)
+    {
+        HashAlgorithm.Initialize();
+        var hashBytes = HashAlgorithm.ComputeHash(bytes);
+        return "sha256-" + Convert.ToBase64String(hashBytes);
+    }
```

Comment:
You can use `Utils.ComputeHash`, and add an overload for `byte[]`

Reply:
I need sha-256 not sha-512. I will move it there.